docs/dev/experimental-mode.adoc

```diff
@@ -19,74 +19,104 @@ Experimental mode can be enabled by setting odo preferences as well.
-  -f, --force-build          Use force-build flag to force building the component
-  -h, --help                 Help for push
-      --ignore stringSlice   Files or folders to be ignored via glob expressions.
-      --show-log             If enabled, logs will be shown when built
-      --source               Use source flag to only push latest source on to cluster
-
-Additional Flags:
-  -v, --v Level              Log level for V logs. Level varies from 0 to 9 (default 0).
-      --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
+Odo Devfile Components:
+NAME                 DESCRIPTION                           SUPPORTED
+maven                Upstream Maven and OpenJDK 11         YES
+nodejs               Stack with NodeJS 10                  YES
+openLiberty          Open Liberty microservice in Java     YES
+java-spring-boot     Spring Boot® using IBM Java           YES
 ```
 
-If the experimental mode is disabled and the hidden flag is used, it will throw an unknown flag error.
-```sh
-$ odo push --devfile ./devfile.yaml  
-Error: unknown flag: --devfile
+##### Creating a devfile component:
+When creating a devfile component with the experimental flag enabled, a local config of the component settings in created at `.odo/env/env.yaml` in the project directory.
+
+```
+$ odo create java-spring-boot myspring
+Experimental mode is enabled, use at your own risk
+
+Validation
+ ✓  Checking Devfile compatibility [234445ns]
+ ✓  Validating Devfile [84112ns]
+
+Please use `odo push` command to create the component with source deployed
+```
+
+##### Creating and deleting component url:
+Create a devfile component url with `odo url create` and delete a devfile component url with `odo url delete`. This updates the component settings at `.odo/env/env.yaml` accordingly.
+
+When creating a component url, the ingress domain must be specified with the flag `--host` and a valid port must be specified with `--port`. The ports to be pushed, must be in the devfile.
+
+```
+$ odo url create myspring --port 8080 --host <ingress domain>
+ ✓  URL created for component: myspring, cluster host: <ingress domain>
+
+To create URL on the OpenShift Cluster, please use `odo push`
+```
+
+When deleting a component url, specify the url name to be deleted
 ```
+$ odo url delete myspring
+? Are you sure you want to delete the url myspring Yes
+ ✓  URL myspring removed from the env file
+
+To delete URL on the OpenShift Cluster, please use `odo push`
+```
+
+To push either of the url sub-command changes, execute `odo push`.
+
+##### Pushing to a devfile component:
+When the experimental mode is enabled, `odo push` creates a pod with all the devfile component containers and volumes. Push syncs the local project on disk to the pod container where devfile component has `mountSources: true`. The optional build and the mandatory run commands are executed in order.
```

Comment:
Nit: It should be `When the experimental mode is enabled and a valid devfile exists` because if no devfile exists and experimental mode is enabled, it will fall back to s2i (provided you're on an openshift cluster)